src/frontend/src/planner/query.rs

```diff
@@ -41,10 +41,6 @@ impl Planner {
         let order = Order {
             column_orders: order,
         };
-        // Optimize order key before using it for TopN.
-        let func_dep = plan.functional_dependency();
-        let order = func_dep.minimize_order_key(order);
-
```

Comment:
Removed. Not clear what's the `dist_key_indices` to pass in here... Only impacts batch top n, we can leave it out for now.

Reply:
We could pass in `[]` as dist_key_indices since top n is singleton dist.

Reply:
Done in https://github.com/risingwavelabs/risingwave/pull/16204/commits/4ee529c12db7bc91bc31deab19918d5b0f3d8e60